types/index.test-d.ts

```diff
@@ -81,3 +85,84 @@ app
     await mysql.execute("SELECT NOW()");
     mysql.connection.end();
   });
+
+app
+  .register(fastifyMysql, {
+    type: "connection",
+    promise: true,
+    connectionString: "mysql://root@localhost/mysql",
+  })
+  .after(async function (err) {
+    const mysql = app.mysql as MySQLPromiseConnection;
```

Comment:
Maybe we should consider implementing typeguards.

Reply:
Do you mean to have something like this
```
function isMySQLPromiseConnection (obj) {
  if (!('connection' in obj)) return false
  return obj && typeof obj.query === 'function' && typeof obj.execute === 'function' && obj.connection && typeof obj.connection === 'object'
}

if (isMySQLPromiseConnection(app.mysql)) {
  const mysql = app.mysql;
  // do stuff
}
```

Reply:
Yeah, and the typescript equivalent in the d.ts

```
declare function isMySQLPromiseConnection (value: unknown): value is MySQLPromiseConnection;
```

untested.

Unfortunately we can not modify FastifyInstance by checking what .register gets as parameters. So maybe it makes sense to add typeguards.